pennylane/math/quantum.py

```diff
@@ -720,6 +720,67 @@ def _compute_mutual_info(
     return vn_entropy_1 + vn_entropy_2 - vn_entropy_12
 
 
+def expectation_value(operator_matrix, state_vector, check_state=False, c_dtype="complex128"):
+    r"""Compute the expectation value of an operator and a pure state.
+
+    The expectation value of an operator :math:`A` for a pure states given by state vector :math:`\ket{\psi}`
+    is defined as
+
+    .. math::
+        \langle A \rangle = \bra{\psi} A \ket{\psi}
+
+    Args:
+        operator_matrix (tensor_like): ``(2**N, 2**N)`` or ``(batch_dim, 2**N, 2**N)`` operator matrix.
+        state_vector (tensor_like): ``(2**N)`` or ``(batch_dim, 2**N)`` state vector.

```

Comment:
Personally preference, but I would put the information _what_ the object is first, and follow up with the shape details :)
```suggestion
        operator_matrix (tensor_like):  operator matrix with shape ``(2**N, 2**N)`` or ``(batch_dim, 2**N, 2**N)``.
        state_vector (tensor_like): state vector with shape ``(2**N)`` or ``(batch_dim, 2**N)``.
```